tests/ResponseTest.php

```diff
@@ -42,52 +41,55 @@ protected function getSimpleMock($classname)
             ->getMock();
     }
 
-    public function testResponceClass()
-    {
-        $this->assertInstanceof(Response::class, $this->response);
-    }
-
-    public function testResponseProtocolDefaultValue()
+    
+    public function testProtocolVersionDefaultValue()
     {
-        $this->assertEquals('1.1', $this->response->getProtocolVersion());
+        $this->assertEquals('1.0', $this->response->getProtocolVersion());
     }
 
-    public function testResponseProtocol()
+    public function testProtocolVersion()
     {
-        $response = $this->response->withProtocolVersion('2.0');
-        $this->assertEquals('2', $response->getProtocolVersion());
+        $response20 = $this->response->withProtocolVersion('2.0');
+        $this->assertEquals('2', $response20->getProtocolVersion());
+        
+        $response2 = $this->response->withProtocolVersion('2');
+        $this->assertEquals('2', $response2->getProtocolVersion());
         
-        $response2 = $this->response->withProtocolVersion('1.1');
-        $this->assertEquals('1.1', $response2->getProtocolVersion());
+        $response11 = $this->response->withProtocolVersion('1.1');
+        $this->assertEquals('1.1', $response11->getProtocolVersion());
         
-        $response3 = $this->response->withProtocolVersion('1.0');
-        $this->assertEquals('1.0', $response3->getProtocolVersion());
+        $response10 = $this->response->withProtocolVersion('1.0');
+        $this->assertEquals('1.0', $response10->getProtocolVersion());
     }
 
-    public function testIstantResponceClass()
+    public function testWithProtocolVersionImmutable()
     {
+        $version = $this->response->getProtocolVersion();
         $response = $this->response->withProtocolVersion('1.1');
+        
         $this->assertInstanceof(Response::class, $response);
-        $this->assertInstanceof(Response::class, $response);
+        $this->assertNotSame($this->response, $response);
```

Comment:
The point of this test is not only that a `Response` object is returned, but also that it's a clone. The original request object has not been modified.